AnbProduct.php

```diff
@@ -130,466 +133,659 @@ function topDealProducts( $atts, $nav = "" ) {
         }
 
-		return $servicesHtml;
-	}
-
-	/**
-	 * @param array $products
-	 *
-	 * @return array
-	 */
-	public function prepareProductsData( array $products ) {
-		$data = [];
-		//for now hardcoded decimal separator to coma
-		foreach ( $products as $idx => $product ) {
-			$data[ $idx ] = $this->prepareProductData( $product );
-		}
-
-		return $data;
-	}
-
-	/**
-	 * @param object $product
-	 *
-	 * @return array
-	 */
-	public function prepareProductData( $product ) {
-		$data = [];
-		//for now hardcoded decimal separator to coma
-		//Pack type: 'int_tel', 'tv_tel', 'int_tel_tv', 'gsm_int_tel_tv', 'int_tv', 'gsm_int_tv'
-		$data['producttype'] = $product->producttype;
-
-		if ( $product->producttype == "packs" ) {
-			$data['packtype'] = $product->packtype;
-			$data['packtypes'] = array_keys((array)$product->packtypes);
-		}
-
-		$data['product_name']  = $product->product_name;
-		$data['sg']            = $product->segment;
-		$data['last_update']   = $product->last_update;
-		$data['product_slug']  = $product->product_slug;
-		$data['supplier_id']   = $product->supplier_id;
-		$data['supplier_slug'] = $product->supplier_slug;
-		$data['supplier_name'] = $product->supplier_name;
-		$data['product_id']    = $product->product_id;
-		$data['tagline']       = isset( $product->texts->tagline ) ? $product->texts->tagline : "";
-		$data['price']         = (array) $product->price;
-		$data['monthly_fee']   = (array) $product->monthly_fee;
-		$data['advantage']     = $product->price->advantage;
-		$data['currency_unit'] = $data['monthly_fee']['unit'];
-		$data['year_1_promo']  = $product->price->year_1_promo;
-        $data['commission']  = $product->commission;
-		//break price into chunks like price, cents and currency
-		$monthlyPrice    = $data['monthly_fee']['value'];
-		$monthlyPriceArr = explode( ".", $monthlyPrice );
-		if ( ! isset( $monthlyPriceArr[1] ) ) {
-			$monthlyPriceArr[1] = 0;
-		}
-		$data['monthly_price_chunk'] = [
-			'price' => $monthlyPriceArr[0],
-			'cents' => ( $monthlyPriceArr[1] < 10 ? '0' . $monthlyPriceArr[1] : $monthlyPriceArr[1] ),
-			'unit'  => $data['monthly_fee']['unit']
-		];
+        return $servicesHtml;
+    }
+
+    /**
+     * @param array $prd
+     *
+     * @return string
+     */
+    function getServicesHtmlBase(array $prd)
+    {
+        $servicesHtml = '';
+        //$prd['packtype']: This is combining mulitple names into one using + sign e.g. Internet + TV
+        $prdOrPckTypes = ($prd['producttype'] == 'packs') ? $prd['packtypes'] : $prd['producttype'];
+        $prdOrPckTypes = (!is_array($prdOrPckTypes)) ? strtolower($prdOrPckTypes) : $prdOrPckTypes;
+
+        if ((is_array($prdOrPckTypes) && in_array('internet', $prdOrPckTypes)) ||
+            (!is_array($prdOrPckTypes) && strpos($prdOrPckTypes, "int") !== false && $prdOrPckTypes != 'mobile_internet')) {
+            $servicesHtml .= '<li class="wifi">
+                                <i class="service-icons wifi print-hide"></i>
```

Comment:
waarom zijn dit geen abf icons?

Reply:
Ik heb alleen die functie gekopieerd met nieuwe name zodat hij dus ook aan te roepen is zonder een extra instance te maken van AnbTopdeals. Aanpassen?